File: GeneratedEntityReaderClasses/Components/CrawlerAnimalComponent.cs
namespace GeneratedNoitaClasses;
public class CrawlerAnimalComponent : NoitaComponentBase,  iNoitaType<CrawlerAnimalComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat ray_length { get; set; } = new();
	public NoitaInt ray_count { get; set; } = new();
	public NoitaFloat gravity { get; set; } = new();
	public NoitaFloat terminal_velocity { get; set; } = new();
	public NoitaFloat speed { get; set; } = new();
	public NoitaInt give_up_area_radius { get; set; } = new();
	public NoitaInt give_up_time { get; set; } = new();
	public NoitaFloat attack_from_ceiling_check_ray_length { get; set; } = new();
	public NoitaInt attack_from_ceiling_check_every_n_frames { get; set; } = new();
	public NoitaFloat collision_damage { get; set; } = new();
	public NoitaFloat collision_damage_radius { get; set; } = new();
	public NoitaInt collision_damage_frames_between { get; set; } = new();
	public NoitaBool animate { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		ray_length.Read(s);
		ray_count.Read(s);
		gravity.Read(s);
		terminal_velocity.Read(s);
		speed.Read(s);
		give_up_area_radius.Read(s);
		give_up_time.Read(s);
		attack_from_ceiling_check_ray_length.Read(s);
		attack_from_ceiling_check_every_n_frames.Read(s);
		collision_damage.Read(s);
		collision_damage_radius.Read(s);
		collision_damage_frames_between.Read(s);
		animate.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		ray_length.Write(s);
		ray_count.Write(s);
		gravity.Write(s);
		terminal_velocity.Write(s);
		speed.Write(s);
		give_up_area_radius.Write(s);
		give_up_time.Write(s);
		attack_from_ceiling_check_ray_length.Write(s);
		attack_from_ceiling_check_every_n_frames.Write(s);
		collision_damage.Write(s);
		collision_damage_radius.Write(s);
		collision_damage_frames_between.Write(s);
		animate.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/CrawlerAnimalComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class CrawlerAnimalComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    ray_length:NoitaFloat = field(default_factory=NoitaFloat)
    ray_count:NoitaInt = field(default_factory=NoitaInt)
    gravity:NoitaFloat = field(default_factory=NoitaFloat)
    terminal_velocity:NoitaFloat = field(default_factory=NoitaFloat)
    speed:NoitaFloat = field(default_factory=NoitaFloat)
    give_up_area_radius:NoitaInt = field(default_factory=NoitaInt)
    give_up_time:NoitaInt = field(default_factory=NoitaInt)
    attack_from_ceiling_check_ray_length:NoitaFloat = field(default_factory=NoitaFloat)
    attack_from_ceiling_check_every_n_frames:NoitaInt = field(default_factory=NoitaInt)
    collision_damage:NoitaFloat = field(default_factory=NoitaFloat)
    collision_damage_radius:NoitaFloat = field(default_factory=NoitaFloat)
    collision_damage_frames_between:NoitaInt = field(default_factory=NoitaInt)
    animate:NoitaBool = field(default_factory=NoitaBool)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.ray_length.Read(s)
        self.ray_count.Read(s)
        self.gravity.Read(s)
        self.terminal_velocity.Read(s)
        self.speed.Read(s)
        self.give_up_area_radius.Read(s)
        self.give_up_time.Read(s)
        self.attack_from_ceiling_check_ray_length.Read(s)
        self.attack_from_ceiling_check_every_n_frames.Read(s)
        self.collision_damage.Read(s)
        self.collision_damage_radius.Read(s)
        self.collision_damage_frames_between.Read(s)
        self.animate.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.ray_length.Write(s)
        self.ray_count.Write(s)
        self.gravity.Write(s)
        self.terminal_velocity.Write(s)
        self.speed.Write(s)
        self.give_up_area_radius.Write(s)
        self.give_up_time.Write(s)
        self.attack_from_ceiling_check_ray_length.Write(s)
        self.attack_from_ceiling_check_every_n_frames.Write(s)
        self.collision_damage.Write(s)
        self.collision_damage_radius.Write(s)
        self.collision_damage_frames_between.Write(s)
        self.animate.Write(s)

